pkg/kubelet/volumemanager/reconciler/reconciler.go

```diff
@@ -586,22 +586,30 @@ func (rc *reconciler) updateDevicePath(volumesNeedUpdate map[v1.UniqueVolumeName
 }
 
 func getDeviceMountPath(volume *reconstructedVolume) (string, error) {
-	volumeAttacher, err := volume.attachablePlugin.NewAttacher()
-	if volumeAttacher == nil || err != nil {
-		return "", err
-	}
-	deviceMountPath, err :=
-		volumeAttacher.GetDeviceMountPath(volume.volumeSpec)
-	if err != nil {
-		return "", err
-	}
-
+	var (
+		deviceMountPath string
+		err             error
+	)
 	if volume.blockVolumeMapper != nil {
+		// for block volume, we return its global map path
 		deviceMountPath, err =
 			volume.blockVolumeMapper.GetGlobalMapPath(volume.volumeSpec)
 		if err != nil {
 			return "", err
 		}
+	} else if volume.attachablePlugin != nil {
+		// for filesystem volume, we return its device mount path if the plugin implements AttachableVolumePlugin
+		volumeAttacher, err := volume.attachablePlugin.NewAttacher()
+		if volumeAttacher == nil || err != nil {
+			return "", err
+		}
+		deviceMountPath, err =
```

Comment:
What about directly `return volumeAttacher.GetDeviceMountPath(volume.volumeSpec)`?

+ the same in `volume.blockVolumeMapper != nil` branch above and you can save quite a few lines of code and a variable.

Reply:
sure, done